### ISSUE TITLE
TS compile errors on 4.10.0 - An index signature parameter type must be 'string' or 'number'.

### ISSUE THREAD

**Issue by timothyallan** [mobxjs/mobx#1986]
Hey all, I'm getting TS compile errors with mobx 4.10.0 and TS 3.4.5 as well as TS 3.5.1 
It's related to https://github.com/mobxjs/mobx/pull/1944

```
../client/node_modules/mobx/lib/types/observableobject.d.ts:37:10 - error TS1023: An index signature parameter type must be 'string' or 'number'.

37         [key: PropertyKey]: ObservableValue<any> | ComputedValue<any>;
            ~~~
```

**Comment by loklaan** [comment]
Hi @timothyallan, could you post an abstract of the code that raises these errors when compiled?

**Comment by timothyallan** [comment]
Yeah I'll look into it some more. It really just was happening while doing a `./node_modules/typescript/bin/tsc -p ./src`, but if nobody else is getting that, there must be something else at play

**Comment by cameronbraid** [comment]
I am getting it to0

**Comment by cameronbraid** [comment]
```
export declare class ObservableObjectAdministration implements IInterceptable<IObjectWillChange>, IListenable {
    target: any;
    name: string;
    defaultEnhancer: IEnhancer<any>;
    values: {
        [key: PropertyKey]: ObservableValue<any> | ComputedValue<any>;
    };
`11

**Comment by cameronbraid** [comment]
Looks like PropertyKey has been changed to include symbol not just string and number

https://github.com/microsoft/TypeScript/blob/master/lib/lib.es5.d.ts#L105

**Comment by loklaan** [comment]
Oh no, I checked it out, looks like the build of the `mobx@4.10.0` is broken. (shown above) 😭
The change that affected this is mine, my bad 😬.

**My recommendation is to stay on `mobx@4.9.4`.**

---

Looks like the implementation in the `mobx4-master` branch uses a keyed object instead of a Map, and Symbols aren't yet supported as object keys (check out #microsoft/TypeScript/1863 for progress). I think the intention was to put a `@ts-ignore` on the line but it gets dropped during the build-step.

Together with #1987, it might be easier to leave out Symbol support for keys for the 4.x release... That's up to the maintainers.

**Comment by mweststrate** [comment]
Fixed in 4.11.0
